lib/effects.js

```diff
@@ -100,6 +49,9 @@ async function hopeThat(callback) {
         result = false
         const msg = err.inspect ? err.inspect() : err.toString()
         debug(`Unsuccessful assertion > ${msg}`)
+        event.dispatcher.on(event.test.finished, test => {
```

Comment:
```suggestion
        event.dispatcher.once(event.test.finished, test => {
```

should be applied for this test only